src/event-stream/event-stream.interfaces.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright © 2022 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ApiProperty } from '@nestjs/swagger';
+
+export interface EventStream {
+  id: string;
+  name: string;
+}
+
+export interface EventStreamSubscription {
+  id: string;
+  name: string;
+  stream: string;
+}
+
+export interface TokenCreateEvent extends Event {
+  data: {
+    operator: string;
```

Comment:
I don't see "operator" in the emit from your factory contract.

Reply:
The journey of "operator" on ERC1155 was that FireFly maps the "key" input parameter to "operator" on the connector, which maps to "fly-from" on ethconnect.
On the way back out, ERC1155 happens to emit `msg.sender` as "operator" in all of its events. We take that and store it as "key" on the FireFly objects when we create them in response to the emitted events.

ERC20 does _not_ appear to emit `msg.sender` in any of its events. I don't know if there's a generic way to look it up via ethconnect... But if we can't reliably lookup the key that signed the input transaction, we may need to drop "key" from some of the token tables. Might need a chat with @peterbroadhurst.

Reply:
One possible route (if we do need to lookup this value) - we already do an extra fetch in ethconnect to `eth_getTransactionByHash` in order to populate `InputMethod` and `InputArgs`. The "from" address is available in that same body, so could enhance ethconnect to populate `InputFrom` at this time as well.

Reply:
yep, in Ethereum, events are side effects of a transaction and are captured in the `logs` section of the transaction receipt, so it's a legitimate approach to parse the transaction that contained the event logs and assign its signer to the events. though the call that obtains the event logs `eth_getFilterLogs` does not include the transaction itself, so it has to be pulled from calling `eth_getTransactionByHash` as ethconnect does for timestamps

Reply:
For now, going to call `GET /abis/<AbiId>/<ContractAdd>/<method>?fly-transaction=<transactionHash>&stream=<streamId>

Reply:
For now, going to send `GET /abis/<AbiId>/<ContractAdd>/<method>?fly-transaction=<transactionHash>&stream=<streamId>` to eth-connect and fetch the operator value